src/components/LinkModal/LinkModal.js

```diff
@@ -0,0 +1,253 @@
+            {i}
+          </option>,
+        );
+      }
+      setDropdownNumbers(numbers);
+
+      //  prepopulate URL if URL is selected
+      const selectedText = core.getSelectedText();
+      if (selectedText) {
+        const urlRegex = /(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/g;
+        const urls = selectedText.match(urlRegex);
+        if (urls && urls.length > 0) {
+          setURL(urls[0]);
+        }
+      }
+    }
+  }, [totalPages, isOpen]);
+
+  useEffect(() => {
+    if (tabSelected === 'PageNumberPanelButton' && isOpen) {
+      pageNumberInput.current.focus();
+    } else if (tabSelected === 'URLPanelButton' && isOpen) {
+      urlInput.current.focus();
+    }
+  }, [tabSelected, isOpen]);
+
+  const modalClass = classNames({
+    Modal: true,
+    LinkModal: true,
+    open: isOpen,
+    closed: !isOpen,
+  });
+
+  return isDisabled ? null : (
+    <div
+      className={modalClass}
+      data-element="linkModal"
+      onMouseDown={closeModal}
+    >
+      <div className="container" onMouseDown={e => e.stopPropagation()}>
+        <Tabs id="linkModal">
+          <div className="header">
+            <div className="tab-list">
+              <Tab dataElement="URLPanelButton">
+                <Button label={t('link.URL')} />
+              </Tab>
+              <Tab dataElement="PageNumberPanelButton">
+                <Button label={t('link.Page')} />
+              </Tab>
+            </div>
+            <ActionButton
+              dataElement="linkModalCloseButton"
+              title="action.close"
+              img="ic_close_black_24px"
+              onClick={closeModal}
+            />
+          </div>
+
+          <TabPanel dataElement="URLPanel">
+            <form onSubmit={addURLLink}>
+              <div className="enter">
+                <div>{t('link.EnterURL')}</div>
+                <input
+                  className="urlInput"
+                  type="url"
+                  ref={urlInput}
+                  value={url}
+                  onChange={e => setURL(e.target.value)}
+                />
+              </div>
+              <div className="buttons">
+                <Button
+                  dataElement="linkSubmitButton"
+                  label={t('action.link')}
+                  onClick={addURLLink}
+                />
+              </div>
+            </form>
+          </TabPanel>
+          <TabPanel dataElement="PageNumberPanel">
+            <form onSubmit={addPageLink}>
+              <div className="enter">
+                <div>{t('link.EnterPage')}</div>
+                <select
+                  className="pageNumberSelect"
+                  ref={pageNumberInput}
+                  value={pageNumber}
+                  onChange={e => setPageNumber(e.target.value)}
+                >
+                  {dropdownNumbers}
+                </select>
+              </div>
+              <div className="buttons">
```

Comment:
Is it necessary to have this `div`? Also I don't think we are using this class name in `LinkModal.scss` so maybe we can just remove it?

Reply:
Resolved